src/main/java/skye/ui/UI.java

```diff
@@ -0,0 +1,195 @@
+
+    public UI() {
+        scanner = new Scanner(System.in);
+    }
+
+    /**
+     * Prints a message to be shown to the user with horizontal lines
+     * before and after the message.
+     *
+     * @param message Message shown to the user
+     */
+    public void printMessage(String message) {
+        renderLine();
+        System.out.println(message);
+        renderLine();
+    }
+
+    /**
+     * Reads the command from the user's input.
+     *
+     * @return A command from the user's input
+     */
+    public String readCommand() {
+        return scanner.nextLine();
+    }
+
+    /**
+     * Renders a horizontal line on the terminal.
+     */
+    public void renderLine() {
+        System.out.println("_".repeat(LINE_LENGTH));
+    }
+
+    /**
+     * Display the welcome message used at the start of the program.
+     */
+    public void showWelcome() {
+        printMessage("Hello! I'm Skye, your personal task assistant.\n"
+                + "What can I do for you?");
+    }
+
+    /**
+     * Display the exit message shown when exiting the program.
+     */
+    public void showGoodBye() {
+        scanner.close();
+        printMessage("Bye. Hope to see you again soon!");
+    }
+
+    /**
+     * Display the error message when writing to the save file has failed.
+     */
+    public void showLoadingError() {
+        printMessage("Sorry! I was unable to load the save file :(");
+    }
+
+    /**
+     * Display the current list of tasks that the user has recorded.
+     *
+     * @param tasks A list of tasks from the TaskList
+     */
+    public void showTasks(List<Task> tasks) {
+        renderLine();
+        System.out.println("Here are the tasks in your list:");
+        if (!tasks.isEmpty()) {
+            for (int i = 0; i < tasks.size(); i++) {
+                System.out.printf("%d.%s%n", i + 1, tasks.get(i));
+            }
+        } else {
+            System.out.println("Nice!! You're all caught up and have no pending tasks to worry about.");
+        }
+        renderLine();
+    }
+
+    /**
+     * Display the task that the user has recently added.
+     *
+     * @param task Task that was added to the task list
+     * @param tasks Task list
+     */
+    public void showAddedTask(Task task, List<Task> tasks) {
+        printMessage(
+            String.format(
+                    "Got it. I've added this task:\n %s\nNow you have %d task(s) in the list.",
+                    task.toString(),
+                    tasks.size()
+            )
+        );
+    }
+
+    /**
+     * Display the task that the user has recently deleted.
+     *
+     * @param task Task that was deleted from the task list
+     * @param tasks Task list
+     */
+    public void showRemovedTask(Task task, List<Task> tasks) {
+        printMessage(
+            String.format(
+                "Noted. I've removed this task:\n %s\nNow you have %d tasks in the list",
+                task.toString(),
+                tasks.size()
```

Comment:
I believe wrapped lines should be indented by 8 spaces?
```suggestion
                    "Noted. I've removed this task:\n %s\nNow you have %d tasks in the list",
                    task.toString(),
                    tasks.size()
```